### ISSUE TITLE
Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集

### ISSUE THREAD

**Issue by zhishao** [YunYang1994/tensorflow-yolov3#442]

**Comment by iodncookie** [comment]
大佬,我也在训练coco2017数据集.我有个问题想请教一下.作者的代码分两阶段训练,我发现开始第二个阶段训练的时候训练集测试集loss会突然上升,然后训练集会慢慢下降,但是测试集下降了几个epoch后就开始上升了,越来越大,请问是我哪里训练有问题么?

**Comment by zhishao** [comment]
@iodncookie 你说的这个我没有遇到，我的test loss目前稳定在27、28左右，train loss一直在降，一直在训练没有停。

**Comment by iodncookie** [comment]
我是按照作者设置的config来运行的,不知道哪里有问题.您的config文件方便发给我看看么?

**Comment by zhishao** [comment]
@iodncookie 我用的也是作者原版config

**Comment by iodncookie** [comment]
@zhishao 方便微信交流一下么.我的微信是 pc544556

**Comment by zhishao** [comment]
@iodncookie 总之就是感觉loss降得比较慢

**Comment by iodncookie** [comment]
@zhishao 你训练一个epoch要多久?

**Comment by zhishao** [comment]
@iodncookie 我用mobilenetv3 batchsize=6 GPU是2070S一个epoch两个小时吧 训练了十天目前train loss 22，test loss 一直是27不变，每个epoch train loss只降0.X，太慢了，不知道是哪里出问题... 然后测试mAP只有百分之零点几...

**Comment by iodncookie** [comment]
应该是加载数据方式的原因.我之前训练也是很慢,后面改了一下加载数据的方式,能快个3倍左右吧.

**Comment by iodncookie** [comment]
我用mobilenet v2训练过,mAP也很低,只有百分之46左右.

**Comment by zhishao** [comment]
@iodncookie 为什么会这么低呢？loss降的慢是什么原因呢？你有分析过吗？是主干网络的原因吗？原版darknet53你试过吗？多少epoch收敛？

**Comment by jhfollowgd** [comment]
> @iodncookie 我用的也是作者原版config

你好，我想问下你们用自己的数据训练了吗，我用自己的数据训练，按照博主的步骤走的，都跑通了，在最后测试的时候，我标记的是“人”这个类别，但是在运行evaluate.py出来的ground-truth跟predicted文件，里面出来的类别是“bird"最后运行main.py出来的也还是”bird"类别的ap，不知道哪里有问题。

**Comment by iodncookie** [comment]
> > @iodncookie 我用的也是作者原版config
> 
> 你好，我想问下你们用自己的数据训练了吗，我用自己的数据训练，按照博主的步骤走的，都跑通了，在最后测试的时候，我标记的是“人”这个类别，但是在运行evaluate.py出来的ground-truth跟predicted文件，里面出来的类别是“bird"最后运行main.py出来的也还是”bird"类别的ap，不知道哪里有问题。

你有改一下coco.names文件么

**Comment by jhfollowgd** [comment]
没有改呢，coconame里面我看有person这个类别就没动，要改为一类吗？------------------ 原始邮件 ------------------
发件人: "Ponchine Tion"<notifications@github.com>
发送时间: 2019年12月18日(星期三) 上午10:31
收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>;
抄送: "jhfollowgd"<1292109155@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442)


  
@iodncookie 我用的也是作者原版config
  
你好，我想问下你们用自己的数据训练了吗，我用自己的数据训练，按照博主的步骤走的，都跑通了，在最后测试的时候，我标记的是“人”这个类别，但是在运行evaluate.py出来的ground-truth跟predicted文&#x4EF6;，里面出来的类别是“bird"最后运行main.py出来的也还是”bird"类别的ap，不知道哪里有问题。
  
你有改一下coco.names文&#x4EF6;么
 
&mdash;

**Comment by iodncookie** [comment]
> 没有改呢，coconame里面我看有person这个类别就没动，要改为一类吗？------------------ 原始邮件 ------------------ 发件人: "Ponchine Tion"<notifications@github.com> 发送时间: 2019年12月18日(星期三) 上午10:31 收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>; 抄送: "jhfollowgd"<1292109155@qq.com>;"Comment"<comment@noreply.github.com>; 主题: Re: [YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442) @iodncookie 我用的也是作者原版config 你好，我想问下你们用自己的数据训练了吗，我用自己的数据训练，按照博主的步骤走的，都跑通了，在最后测试的时候，我标记的是“人”这个类别，但是在运行evaluate.py出来的ground-truth跟predicted文&#x4EF6;，里面出来的类别是“bird"最后运行main.py出来的也还是”bird"类别的ap，不知道哪里有问题。 你有改一下coco.names文&#x4EF6;么 &mdash; You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

嗯嗯,要改一下

**Comment by jhfollowgd** [comment]
我把coconame改为一类运行报错，然后就把80类全改为80个person运行出来了。这是不是要改其他地方的代码啊？😂😂------------------ 原始邮件 ------------------
发件人: "Ponchine Tion"<notifications@github.com>
发送时间: 2019年12月18日(星期三) 中午11:11
收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>;
抄送: "jhfollowgd"<1292109155@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442)


 
没有改呢，coconame里面我看有person这个类别就没动，要改为一类吗？------------------ 原始邮&#x4EF6; ------------------ 发&#x4EF6;人: "Ponchine Tion"notifications@github.com 发送时间: 2019年12月18日(星期三) 上午10:31 收&#x4EF6;人: "YunYang1994/tensorflow-yolov3"tensorflow-yolov3@noreply.github.com; 抄送: "jhfollowgd"1292109155@qq.com;"Comment"comment@noreply.github.com; 主题: Re: [YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442) @iodncookie 我用的也是作者原版config 你好，我想问下你们用自己的数据训练了吗，我用自己的数据训练，按照博主的步骤走的，都跑通了，在最后测试的时候，我标记的是“人”这个类别，但是在运行evaluate.py出来的ground-truth跟predicted文&#x4EF6;，里面出来的类别是“bird"最后运行main.py出来的也还是”bird"类别的ap，不知道哪里有问题。 你有改一下coco.names文&#x4EF6;么 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
  
嗯嗯,要改一下
 
&mdash;

**Comment by iodncookie** [comment]
一旦类别固定了,网络最后的输出也就固定了,所以你要改类别的话得重新训练了.训练代码不用改,只要把类别文件改了就行.

**Comment by iodncookie** [comment]
@jhfollowgd

**Comment by jhfollowgd** [comment]
嗯嗯，那我这样改成80个类别直接测试有问题吗？还是需要改了类别再重新训练吗？还有个问题（可能也是我没训练好吧）就是训练得到的模型，需要转为相应的pb文件然后在imagedemo里面测试吗？我虽然现在检查精度有95，但是在imagedemo里面运行，图片什么类别也检测不出来。------------------ 原始邮件 ------------------
发件人: "Ponchine Tion"<notifications@github.com>
发送时间: 2019年12月18日(星期三) 中午11:33
收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>;
抄送: "jhfollowgd"<1292109155@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442)



一旦类别固定了,网络最后的输出也就固定了,所以你要改类别的话得重新训练了.训练代码不用改,只要把类别文&#x4EF6;改了就行.
 
&mdash;

**Comment by jhfollowgd** [comment]
Coco的数据我没训练，但是给的demo应该是9。个位数字吧。------------------ 原始邮件 ------------------
发件人: "Ponchine Tion"<notifications@github.com>
发送时间: 2019年12月18日(星期三) 中午11:36
收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>;
抄送: "jhfollowgd"<1292109155@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442)



@jhfollowgd
 
&mdash;

**Comment by iodncookie** [comment]
个人感觉还是要重新训练的,毕竟网络最后预测的输出跟类别有关,会导致训练过程有一定区别. 测试的话需要将模型转化为pb文件. @jhfollowgd

**Comment by jhfollowgd** [comment]
嗯嗯，非常感谢您！------------------ 原始邮件 ------------------
发件人: "Ponchine Tion"<notifications@github.com>
发送时间: 2019年12月18日(星期三) 中午11:50
收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>;
抄送: "jhfollowgd"<1292109155@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442)



个人感觉还是要重新训练的,毕竟网络最后预测的输出跟类别有关,会导致训练过程有一定区别. 测试的话需要将模型转化为pb文&#x4EF6;. @jhfollowgd
 
&mdash;

**Comment by jhfollowgd** [comment]
你好，想问个问题，在运行时，batch_size作者是设置了6，你有修改这些大小吗？我修改了几次，运行完第一阶段就停止了，第二阶段运行不了。 
报这个错，好像是说内存不够了。
I  tensorflow/core/common_runtime/bfc_allocator.cc:645] Chunk at 0000000217715F00 of size 1024

**Comment by jersonal** [comment]
> 你好，想问个问题，在运行时，batch_size作者是设置了6，你有修改这些大小吗？我修改了几次，运行完第一阶段就停止了，第二阶段运行不了。
> 报这个错，好像是说内存不够了。
> I tensorflow/core/common_runtime/bfc_allocator.cc:645] Chunk at 0000000217715F00 of size 1024

我也遇到这样的问题，第二阶段优化全部层，所以显存溢出了，我改动batchsize让其在第二阶段可以正常运行后，重新训练了。

**Comment by Shank2358** [comment]
请问你们在COCO上训练出来了吗，我的val loss一直在二十多就下不去了，没能完全复现paper的结果不过也差不多了。mAP0.5=51左右。
用自己的数据darknet53训练的效果还不错mAP=76左右，不过换成mobilenetv2之后mAP实验了几次都只有45~50，讲道理这俩模型不应该差这么多啊。请问是和调参有关系吗。

**Comment by iodncookie** [comment]
@Shank2358 mobilenetv2的精度应该比darknet53小一些.调参帮助应该不大.

**Comment by jhfollowgd** [comment]
我想问下大家,你们知道日志怎么打开吗?DESKTOP-OQHVMV6这种文件怎么打开,想用日志画一个loss损失函数的迭代图!

**Comment by CNUyue** [comment]
您好，我也遇到了loss不下降的问题，基本从第二个epoch之后loss就稳定了，train_loss会稍微降一点，但是test_loss基本不降，还有就是训练完，eval的结果ap只有零点几，这怎么解决呢@zhishao@iodncookie

**Comment by ylqi007** [comment]
@zhishao @iodncookie @jhfollowgd @CNUyue 
你好，想问下你们的 COCO 数据集是怎么处理然后用于训练的？是把 COCO 转换成 PASCAL VOC 的格式吗？

谢谢

**Comment by iodncookie** [comment]
是的，仓主项目里有转化文件。



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: ylqi007 <notifications@github.com&gt;
发送时间: 2020年10月20日 09:19
收件人: YunYang1994/tensorflow-yolov3 <tensorflow-yolov3@noreply.github.com&gt;
抄送: Ponchine Tion <641978569@qq.com&gt;, Mention <mention@noreply.github.com&gt;
主题: 回复：[YunYang1994/tensorflow-yolov3] Trainning on coco2017, how much can you stop training when the loss value drops. 损失值降到多少时可以停止训练？使用COCO2017数据集 (#442)





 
@zhishao @iodncookie @jhfollowgd @CNUyue
 你好，想问下你们的 COCO 数据集是怎么处理然后用于训练的？是把 COCO 转换成 PASCAL VOC 的格式吗？
 
谢谢

**Comment by jersonal** [comment]
> @zhishao @iodncookie @jhfollowgd @CNUyue @Shank2358 @jersonal
> 你好，想问下你们的 COCO 数据集是怎么处理然后用于训练的？是把 COCO 转换成 PASCAL VOC 的格式吗？
> 
> 谢谢

将coco数据集转换成这个项目需要的格式,该项目的readme中有写,转换代码需要自己写.
